packages/block-editor/src/components/inspector-controls/fill.js

```diff
@@ -36,7 +39,29 @@ export default function InspectorControlsFill( {
 		group = __experimentalGroup;
 	}
 
-	const context = useBlockEditContext();
+	const { clientId, ...context } = useBlockEditContext();
+
+	// Register any clientIds with `contentOnly` controls before the
+	// `mayDisplayControls` early return so that parent blocks can know whether
+	// child blocks have those controls.
+	const { addContentOnlyControlsBlock, removeContentOnlyControlsBlock } =
+		unlock( useDispatch( blockEditorStore ) );
+	useEffect( () => {
```

Comment:
This feels a bit weird to me (two-way syncing). Could we instead calculate the `contentOnly` blocks using a memoed selector? I think this would mean keeping a static list of supported blocks though 🤔.